plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/fs/DBNPathBase.java

```diff
@@ -289,7 +287,20 @@ public void dropNodes(DBRProgressMonitor monitor, Collection<DBNNode> nodes) thr
                     break;
                 }
                 Path resource = node.getAdapter(Path.class);
-                if (resource == null || !Files.exists(resource)) {
+                if (resource == null) {
+                    monitor.subTask("Copy file");
+                    try {
+                        InputStream adapter = node.getAdapter(InputStream.class);
+                        if (adapter != null) {
+                            String fileName = node.getNodeDisplayName();
+                            copyInputStream(adapter, folder, fileName);
```

Comment:
... after you apply the previous suggestion, this method can be inlined. It doesn't make the code any cleaner.

Just a nitpick for the future: don't pass redundant arguments when not required. Here, you pass `folder` and `fileName` just to `folder.resolve(fileName)` later in that method. You could've passed the resulting `Path` as a single argument there.